[brain] numpy.ndarray.__or__ is binary

### Steps to reproduce
Lint following file:
```
import numpy

class BetterNDArray(numpy.ndarray):
  def __or__(self, other):
    return super().__or__(other)
```

This is likely caused by https://github.com/PyCQA/astroid/blob/master/astroid/brain/brain_numpy_ndarray.py#L74

### Current behavior
```
************* Module numpy_ndarray
numpy_ndarray.py:4:2: W0235: Useless super delegation in method '__or__' (useless-super-delegation)
numpy_ndarray.py:5:11: E1121: Too many positional arguments for method call (too-many-function-args)
```

### Expected behavior
No E1121 error.

### ``python -c "from astroid import __pkginfo__; print(__pkginfo__.version)"`` output
2.4.1

@eugene57 thanks for the report.
You are right, the ``__or__`` method is a binary operator. The PR #832 corrects this and now the result of ``pylint`` on your piece of code is the following:
```
bug_astroid_815.py:5:4: W0235: Useless super delegation in method '__or__' (useless-super-delegation)

------------------------------------------------------------------
Your code has been rated at 7.50/10 (previous run: 7.50/10, +0.00)
```
which is, IMO, correct.

--- a/ChangeLog
+++ b/ChangeLog
@@ -7,6 +7,11 @@ What's New in astroid 2.5.0?
 ============================
 Release Date: TBA
 
+* Fixes a bug in the signature of the ``ndarray.__or__`` method, 
+  in the ``brain_numpy_ndarray.py`` module.
+
+  Fixes #815
+
 * Added a brain for ``sqlalchemy.orm.session``
 
 * Separate string and bytes classes patching

--- a/astroid/brain/brain_numpy_ndarray.py
+++ b/astroid/brain/brain_numpy_ndarray.py
@@ -71,7 +71,7 @@ def infer_numpy_ndarray(node, context=None):
         def __mul__(self, value): return numpy.ndarray([0, 0])
         def __ne__(self, value): return numpy.ndarray([0, 0])
         def __neg__(self): return numpy.ndarray([0, 0])
-        def __or__(self): return numpy.ndarray([0, 0])
+        def __or__(self, value): return numpy.ndarray([0, 0])
         def __pos__(self): return numpy.ndarray([0, 0])
         def __pow__(self): return numpy.ndarray([0, 0])
         def __repr__(self): return str()

